src/back/msl/writer.rs

```diff
@@ -401,13 +400,25 @@ impl crate::AddressSpace {
     /// passed through any functions called from the entry point.
     fn needs_pass_through(&self) -> bool {
         match *self {
-            crate::AddressSpace::Uniform
-            | crate::AddressSpace::Storage { .. }
-            | crate::AddressSpace::Private
-            | crate::AddressSpace::WorkGroup
-            | crate::AddressSpace::PushConstant
-            | crate::AddressSpace::Handle => true,
-            crate::AddressSpace::Function => false,
+            Self::Uniform
+            | Self::Storage { .. }
+            | Self::Private
+            | Self::WorkGroup
+            | Self::PushConstant
+            | Self::Handle => true,
+            Self::Function => false,
+        }
+    }
+
+    /// Returns true if the address space may need a "const" qualifier.
```

Comment:
I'm in the dark as to why, say, `Uniform` and `PushConstant` don't need `const`. Is it because their MSL address space alone implies that they're not writable, so `const` is unnecessary, unlike `Storage`, `Private`, and `WorkGroup`, which could be writable?

Reply:
Added a comment here, and made the MSL output actually contain push constants for the test, in #1767